rpc/ExperimentalConsumer.ts

```diff
@@ -0,0 +1,245 @@
+  blockHashPendings: Deferred<string>[] = []
+  blockHash(blockNumber?: number) {
+    if (typeof blockNumber === "number") return this.archiveConsumer.blockHash(blockNumber)
+    const pending = deferred<string>()
+    this.blockHashPendings.push(pending)
+    return pending
+  }
+
+  blockPendings: Deferred<known.SignedBlock>[] = []
+  block(blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.block(blockHash)
+    const pending = deferred<known.SignedBlock>()
+    this.blockPendings.push(pending)
+    return pending
+  }
+  blockFlush(followId: string, blockHash: string) {
+    if (this.blockPendings.length) {
+      const controller = new AbortController()
+      // TODO: why not emitting?
+      this.subscription("chainHead_unstable_body", undefined, [followId, blockHash], (_result) => {
+        while (this.blockPendings.length) {
+          const blockPending = this.blockPendings.shift()!
+          blockPending.resolve(null!)
+        }
+      }, controller.signal)
+    }
+  }
+
+  extrinsicsPendings: Deferred<Uint8Array>[] = []
+  extrinsics(blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.block(blockHash)
+    const pending = deferred<Uint8Array>()
+    this.extrinsicsPendings.push(pending)
+    return pending
+  }
+  extrinsicsFlush(followId: string, blockHash: string) {
+    if (this.extrinsicsPendings.length) {
+      const controller = new AbortController()
+      this.subscription<{ event: "done"; result: string }>(
+        "chainHead_unstable_body",
+        undefined,
+        [followId, blockHash],
+        (result) => {
+          while (this.extrinsicsPendings.length) {
+            const blockPending = this.extrinsicsPendings.shift()!
+            blockPending.resolve(hex.decode(result.result))
+          }
+        },
+        controller.signal,
+      )
+    }
+  }
+
+  keys(key: Uint8Array, limit: number, start?: Uint8Array, blockHash?: string) {
+    return this.archiveConsumer.keys(key, limit, start, blockHash)
+  }
+
+  valuesKeys: Record<string, true> = {}
+  valuesPendings: [keys: string[], pending: Deferred<Uint8Array[]>][] = []
+  values(keys: Uint8Array[], blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.values(keys, blockHash)
+    const keysEncoded = keys.map((key) => {
+      const encoded = hex.encodePrefixed(key)
+      this.valuesKeys[encoded] = true
+      return encoded
+    })
+    const pending = deferred<Uint8Array[]>()
+    this.valuesPendings.push([keysEncoded, pending])
+    return pending
+  }
+  valuesFlush(followId: string, blockHash: string) {
+    const keys = Object.entries(this.valuesKeys)
+    if (keys.length) {
+      const items = keys.map((key) => ({ key, type: "value" }))
+      const controller = new AbortController()
+      this.subscription<{ event: "items"; items: { key: string; value: string }[] }>(
+        "chainHead_unstable_storage",
+        "chainHead_unstable_stopStorage",
+        [followId, blockHash, items, null],
```

Comment:
I'm not sure if the docs are outdated or if the `polkadot`/`substrate` API implementation is not complete.

Passing a single hex encoded key works.

Something like 
```
      this.subscription<{ event: "items"; items: { key: string; value: string }[] }>(
        "chainHead_unstable_storage",
        "chainHead_unstable_stopStorage",
        [followId, blockHash, items[0]?.key[0], null],
        (message) => {
```

And the response is something like

```
<<< {
  message: {
    jsonrpc: "2.0",
    method: "chainHead_unstable_storage",
    params: {
      subscription: "x1sMLqhwCWeg7QBJ",
      result: {
        event: "done",
        result: "0x000000000000000001000000000000004035323d3400000000000000000000000000000000000000000000000000000000"... 62 more characters
      }
    }
  }
}
```

It seems that the `chainHead_unstable_storage` is working for single key reads

Reply:
Investigated this API spec updates and the multiple item support was documented 3 weeks ago
See https://github.com/paritytech/json-rpc-interface-spec/commit/bea1a60a37f15791f51d82719538b27e9789a68d

Smoldot hasn't been updated as it supports a single item request
See https://github.com/paritytech/smoldot/blob/938055a638ec201c022f680c8e8cbd0349e70ed1/bin/light-base/src/json_rpc_service/chain_head.rs#L874

And I guess `substrate` is in the same state